adapter/pkg/operator/synchronizer/gateway_synchronizer.go

```diff
@@ -59,8 +59,8 @@ func HandleGatewayLifeCycleEvents(ch *chan GatewayEvent) {
 // deployGateway deploys the related Gateway in CREATE and UPDATE events.
 func deployGateway(gatewayState GatewayState, state string) error {
 	var err error
-	if gatewayState.GatewayDefinition != nil {
-		_, err = AddOrUpdateGateway(gatewayState.GatewayDefinition, state)
+	if gatewayState.GatewayDefinition != nil || gatewayState.ResolvedListenerCerts != nil {
+		_, err = AddOrUpdateGateway(gatewayState.GatewayDefinition, gatewayState.ResolvedListenerCerts, state)
```

Comment:
shall we input `gatewayState` here instead of `gatewayState.GatewayDefinition` which includes all the data that will be used including listener certs?

Reply:
fixed